### PR TITLE
Fix permission prompt's width and height

### PR DESCRIPTION
**Pull request by ayaNader**
This PR adjusts the width and height for the permission prompt asking user to allow third party to access the Smart Card Connector.

Previously the width was large compared to that in the app mode and the height was too small that a scroll bar was displayed...which was unnecessary since the window height overall isn't big enough to require a scroll bar.

### CODE REVIEW

File: common/js/src/popup-window/in-popup-main-script.js [hunk 1]
```diff
@@ -43,6 +43,8 @@ const GSC = GoogleSmartCard;
 /** @type {!goog.log.Logger} */
 const logger = GSC.Logging.getScopedLogger('PopupWindow.InPopupMainScript');
 
+const titleBarHeight = 35;
```
Comment (emaxx-google) [inline common/js/src/popup-window/in-popup-main-script.js:46]: nit: `TITLE_BAR_HEIGHT`
(See https://google.github.io/styleguide/jsguide.html#naming-constant-names)

File: common/js/src/popup-window/in-popup-main-script.js [hunk 2]
```diff
@@ -155,7 +157,7 @@ GSC.InPopupMainScript.setWindowHeightToFitContent = function() {
     chrome.app.window.current().innerBounds.height = wholeContentHeight;
   } else if (GSC.Packaging.MODE === GSC.Packaging.Mode.EXTENSION) {
     chrome.windows.update(
-        chrome.windows.WINDOW_ID_CURRENT, {'height': wholeContentHeight});
+        chrome.windows.WINDOW_ID_CURRENT, {'height': wholeContentHeight + titleBarHeight});
```
Comment (emaxx-google) [inline common/js/src/popup-window/in-popup-main-script.js:160]: nit: Please add a comment explaining why we need to do that - so the code reader can understand why we do it here and why we don't do it in the "if (GSC.Packaging.MODE === GSC.Packaging.Mode.APP)" case above.

File: common/js/src/popup-window/popup-opener.js [hunk 3]
```diff
@@ -98,9 +100,7 @@ GSC.PopupOpener.createWindow = function(url, windowOptions, opt_data) {
             'frame': windowOptions['frame'],
             'hidden': windowOptions['hidden'],
             'id': windowOptions['id'],
-            'innerBounds': {
```
Comment (emaxx-google) [inline common/js/src/popup-window/popup-opener.js:101]: Is this change intentional? I suppose the old code was actually correct, since `chrome.app.window.create()` takes the "innerBounds" parameter.

File: common/js/src/popup-window/popup-opener.js [hunk 4]
```diff
@@ -62,6 +62,8 @@ const DEFAULT_DIALOG_CREATE_WINDOW_OPTIONS = {
   'visibleOnAllWorkspaces': true
 };
 
+const dialogWindowWidth = 500;
```
Comment (emaxx-google) [inline common/js/src/popup-window/popup-opener.js:65]: I don't think this constant is needed - first, because the callers provide the "width" parameter, and, second, it's conceptually wrong to hardcode the width in this common file.

File: third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompt-dialog.html [hunk 5]
```diff
@@ -34,8 +34,6 @@
 <body>
   <h1 id="header"></h1>
 
-  <button id="close-window">×</button>
```
Comment (emaxx-google) [inline third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompt-dialog.html:37]: The changes related to the "close" button should better be extracted into a separate PR, I think.

Also we still need this button in the app PACKAGING mode, so we shouldn't just delete it. (A probable solution could be to check the PACKAGING mode in the JS code and make the button invisible when the mode is !=app.)

**Review by emaxx-google — COMMENTED** [review]
Thanks.

File: common/js/src/popup-window/popup-opener.js (hunk 4 above)
Comment (ayaNader) [reply common/js/src/popup-window/popup-opener.js:65]: This was a workaround because the default size is bigger than that in the APP mode.

Comment (emaxx-google) [reply common/js/src/popup-window/popup-opener.js:65]: The root cause is here I think:
https://github.com/GoogleChromeLabs/chromeos_smart_card_connector/blob/master/third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompting-checker.js#L55

(We need to pass `'width'` without nesting inside `'innerBounds'`)

File: common/js/src/popup-window/in-popup-main-script.js [hunk 6]
```diff
@@ -154,8 +156,12 @@ GSC.InPopupMainScript.setWindowHeightToFitContent = function() {
   if (GSC.Packaging.MODE === GSC.Packaging.Mode.APP) {
     chrome.app.window.current().innerBounds.height = wholeContentHeight;
   } else if (GSC.Packaging.MODE === GSC.Packaging.Mode.EXTENSION) {
+    // Adding TITLE_BAR_HEIGHT as it's not included in the calculation of the
+    // offsetHeight..this is not needed in APP mode as it doesn't have a
```
Comment (emaxx-google) [inline common/js/src/popup-window/in-popup-main-script.js:160]: nit: Typos: s/offsetHeight..this/offsetHeight. This/

File: common/js/src/popup-window/popup-opener.js (hunk 4 above)
Comment (emaxx-google) [reply common/js/src/popup-window/popup-opener.js:65]: After fixing that place, the `dialogWindowWidth` constant could be removed I believe.

**Review by emaxx-google — CHANGES_REQUESTED** [review]

**Review by emaxx-google — APPROVED** [review]
LGTM

File: third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompting-checker.js [hunk 7]
```diff
@@ -52,7 +52,7 @@ const USER_PROMPT_DIALOG_URL =
     'pcsc_lite_server_clients_management/user-prompt-dialog.html';
 
 const USER_PROMPT_DIALOG_WINDOW_OPTIONS_OVERRIDES = {
-  'innerBounds': {'width': 300}
+  'width': 500
```
Comment (emaxx-google) [inline third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompting-checker.js:55]: Why was the constant changed from 300 to 500? Was it intentional?

Comment (ayaNader) [reply third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompting-checker.js:55]: I remember from our offline discussion that 500 was the correct width..and this is actually the width I always got when testing both app and extension mode...but looking back through previous commits "300" seems to be the original width indeed so will create another PR for that

Comment (emaxx-google) [reply third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompting-checker.js:55]: Hmm, I don't remember us discussing the need to increase the width... If you can help me recall the reason for increasing the width let's discuss it. Otherwise, yup, let's revert this to use 300 pixels as it was before.
